internal/gps/result.go

```diff
@@ -61,19 +65,45 @@ func WriteDepTree(basedir string, l Lock, sm SourceManager, sv bool) error {
 		return err
 	}
 
-	// TODO(sdboyer) parallelize
-	for _, p := range l.Projects() {
-		to := filepath.FromSlash(filepath.Join(basedir, string(p.Ident().ProjectRoot)))
+	projects := l.Projects()
+	worksz := len(projects)
+
+	workChan := make(chan LockedProject, worksz)
+	doneChan := make(chan error, worksz)
```

Comment:
let's create some local tuple type here for this message - something like

```go
type done struct {
	root ProjectRoot
	err error
}
```

so that we don't have to shoehorn all that information into the error, and we can pass back up the lower-level errors unmodified.